### PR TITLE
[Bug Fix]LPS-72939 and LPS-79508

### PR DESCRIPTION
**Pull request by tinatian**
Hi @slnn, 
could you please run a diff with this pull?

Hi @arthurchan35, 
please see my changes, if you have any question, please feel free to let me know.

cc @samuelkong

Thanks.
Tina.

### CODE REVIEW

**Comment by tinatian** [comment]
ci:test

**Comment by slnn** [comment]
@tinatian 

OK, got it.

File: portal-impl/src/com/liferay/portal/util/HttpImpl.java [hunk 1]
```diff
@@ -645,13 +645,27 @@ public String getProtocol(String url) {
 			return url;
 		}
 
+		url = url.trim();
```
Comment (shuyangzhou) [inline portal-impl/src/com/liferay/portal/util/HttpImpl.java]: This trim is wrong. With this if I do:
```
Assert.assertEquals("https", _httpImpl.getProtocol(" https://foo.com"));
```
I will get:
```
expected:<[https]> but was:<[ http]>
```

**Comment by shuyangzhou** [comment]
See my comment at https://github.com/shuyangzhou/liferay-portal/pull/6014/commits/b6c8369425b8c0192e11391925f41146a776262d#r184475390

**Comment by tinatian** [comment]
Sorry, my bad, see https://github.com/shuyangzhou/liferay-portal/pull/6033

**Pull request closed without merging**